src/NCalc.Core/Parser/LogicalExpressionParser.cs

```diff
@@ -22,7 +26,50 @@ public static class LogicalExpressionParser
 
     private const string InvalidTokenMessage = "Invalid token in expression";
 
+    /// <summary>
+    /// Creates or retrieves a cached expression parser with the specified options.
+    /// </summary>
+    /// <param name="options">The parser options containing culture info and argument separator.</param>
+    /// <returns>A parser configured with the specified options.</returns>
+    public static Parser<LogicalExpression> GetOrCreateExpressionParser(LogicalExpressionParserOptions options)
+    {
+        return ParserCache.GetOrAdd(options, CreateExpressionParser);
+    }
+
+    /// <summary>
+    /// Converts an ArgumentSeparator enum value to its corresponding character.
+    /// </summary>
+    /// <param name="separator">The ArgumentSeparator enum value.</param>
+    /// <returns>The character representation of the separator.</returns>
+    /// <exception cref="ArgumentOutOfRangeException">Thrown when the separator value is not a valid ArgumentSeparator enum value.</exception>
+    private static char GetSeparatorChar(ArgumentSeparator separator)
+    {
+        return separator switch
+        {
+            ArgumentSeparator.Semicolon => ';',
+            ArgumentSeparator.Colon => ':',
+            ArgumentSeparator.Comma => ',',
+            _ => throw new ArgumentOutOfRangeException(nameof(separator), separator,
+                $"Unhandled ArgumentSeparator value: {separator}")
+        };
+    }
+
+    /// <summary>
+    /// Creates a new expression parser with the specified options.
+    /// </summary>
+    /// <param name="options">The parser options containing culture info and argument separator.</param>
+    /// <returns>A new parser configured with the specified options.</returns>
+    private static Parser<LogicalExpression> CreateExpressionParser(LogicalExpressionParserOptions options)
+    {
+        return CreateExpressionParser(options.CultureInfo, GetSeparatorChar(options.ArgumentSeparator));
+    }
+
     private static Parser<LogicalExpression> CreateExpressionParser(CultureInfo cultureInfo)
+    {
+        return CreateExpressionParser(cultureInfo, ','); // Default comma separator
+    }
+
+    private static Parser<LogicalExpression> CreateExpressionParser(CultureInfo cultureInfo, char argumentSeparator)
```

Comment:
And then, we will no longer need CultureInfo overloads, because we will have an implicit conversion.